Sources/Linq2DynamoDb.DataContext/EntityWrapper.cs

```diff
@@ -14,6 +16,31 @@ internal class EntityWrapper : IEntityWrapper
         private readonly IEntityKeyGetter _keyGetter;
         private Document _doc, _newDoc;
 
+        private PropertyInfo _entityVersionNumberProperty;
+        private bool _hasResolvedEntityVersionNumberProperty;
+
+        /// <summary>
+        /// Gets PropertyInfo for the entity's property that has [DynamoDBVersion] 
+        /// attribute or returns null if there is none.
+        /// </summary>
+        private PropertyInfo EntityVersionNumberProperty {
+            get {
+                if (!_hasResolvedEntityVersionNumberProperty) {
+                    _entityVersionNumberProperty = Entity
+                        .GetType()
+                        .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                        .Where(property =>
+                            property
+                                .GetCustomAttributes(typeof(DynamoDBVersionAttribute), true)
+                                .SingleOrDefault() != null
+                        ).SingleOrDefault();
```

Comment:
you can collapse that `GetProperties(...).Where(predicate).SingleOrDefault()` into `GetProperties(...)SingleOrDefault(predicate)`